src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresAccessPolicyDAO.scala

```diff
@@ -744,23 +744,53 @@ class PostgresAccessPolicyDAO(protected val writeDbRef: DbReference, protected v
   override def deleteAllResourcePolicies(resourceId: FullyQualifiedResourceId, samRequestContext: SamRequestContext): IO[Unit] = {
     val p = PolicyTable.syntax("p")
     val g = GroupTable.syntax("g")
+    val gm = GroupMemberTable.syntax("gm")
 
-    serializableWriteTransaction("deletePolicy", samRequestContext)({ implicit session =>
```

Comment:
moved this write transaction into a for comp.
this write transaction was split into 1 read transaction and 1 write transaction. this is so that we can still use groupPKsToDelete if the write transaction fails